Finish KIC plugin pages (#1231)
Kong/developer.konghq.com: app/_how-tos/kic-plugin-oidc.md

---

> title: OIDC

Reviewer: ```suggestion
title: Authenticate requests with OIDC and KIC
```

description: "Authenticate requests using the OpenID Connect protocol and {{ site.base_gateway }}"
content_type: how_to

permalink: /kubernetes-ingress-controller/oidc/
breadcrumbs:
  - /kubernetes-ingress-controller/
  - index: kubernetes-ingress-controller
    section: How To

> products:

Reviewer: ```suggestion
plugins:
  - oidc
  
products:
```

  - kic

tools:
  - kic

works_on:
  - on-prem
  - konnect

entities: []

tldr:
  q: How do I configure the OpenID Connect (OIDC) plugin with {{ site.kic_product_name }}?

>   a: Create a `KongPlugin` instance containing your `client_id`, `client_secret` and `grant_type` then annotate a Service or Route with `konghq.com/plugins=my-oidc-plugin`

Reviewer: ```suggestion
  a: Create a `KongPlugin` instance containing your `client_id`, `client_secret`, and `grant_type`, then annotate a Service or Route with `konghq.com/plugins=my-oidc-plugin`.
```

prereqs:
  enterprise: true
  kubernetes:
    gateway_api: true
  entities:
    services:
      - echo-service
    routes:
      - echo
cleanup:
  inline:
    - title: Uninstall KIC from your cluster
      include_content: cleanup/products/kic
      icon_url: /assets/icons/kubernetes.svg
---

> ## About OpenID Connect
> 
> {{ site.base_gateway }} Enterprise’s OIDC plugin can authenticate requests using OpenID Connect protocol. Learn how to setup the OIDC plugin using the {{ site.kic_product_name }}.
> 
> {% include prereqs/kubernetes/keycloak.md render_inline=true %}
> 
> ## Configure the OpenID Connect plugin
> 
> This example uses `keycloak.$PROXY_IP.nip.io` as the host, but you can use any domain name of your choice. For demo purposes, you can use the nip.io service to avoid setting up a DNS record.

Reviewer: ```suggestion

{% include prereqs/kubernetes/keycloak.md render_inline=true %}

## 1. Configure the OpenID Connect plugin

{{ site.base_gateway }}'s [OIDC](/plugins/openid-connect/) plugin can authenticate requests using the OpenID Connect protocol. 

This example uses `keycloak.$PROXY_IP.nip.io` as the host, but you can use any domain name of your choice. For demo purposes, you can use the nip.io service to avoid setting up a DNS record.
```

Reviewer: I got this error from the Keycloak route step:

```
The HTTPRoute "keycloak" is invalid: spec.hostnames[0]: Invalid value: "keycloak..nip.io": spec.hostnames[0] in body should match '^(\*\.)?[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$'
```

Author: This is because `PROXY_IP` is not set

{% entity_example %}
type: plugin
data:
  name: openid-connect
  config:
    issuer: https://keycloak.$PROXY_IP.nip.io/realms/master
    client_id:
    - $CLIENT_ID
    client_secret:
    - $CLIENT_SECRET
    redirect_uri:
    - http://$PROXY_IP/echo

  service: echo
{% endentity_example %}

> ## Validate your configuration

Reviewer: ```suggestion
## 2. Validate your configuration
```

> Once the resource has been reconciled, you'll be able to call the `/echo` endpoint and {{ site.base_gateway }} will route the request to the `echo` service.

Reviewer: ```suggestion
Once the resource has been reconciled, you'll be able to call the `/echo` endpoint and {{ site.base_gateway }} will route the request to the `echo` Service.
```
